app/src/main/java/com/example/composetemplate/di/AppModule.kt

```diff
@@ -1,8 +1,9 @@
 package com.example.composetemplate.di
 
 import com.example.composetemplate.MainApplication
+import org.koin.android.ext.koin.androidApplication
 import org.koin.dsl.module
 
 val appModule = module {
-    single { MainApplication() }
+    single { androidApplication() as? MainApplication }
```

Comment:
Yes, I was getting a NullPointerException when I used application.applicationContext to initialize the DS. I searched the internet and here's what I found out:

The NullPointerException issue you're encountering with application.applicationContext being null in your MainViewModel is likely due to the way the MainApplication instance is being provided in your appModule. When you define single { MainApplication() }, it creates a new instance of MainApplication instead of using the already created instance by the Android framework. The new instance doesn't have its applicationContext set properly, leading to the null reference.

Instead of providing MainApplication directly, you should provide the Application context that is already created by the Android framework. Here's how you can modify your setup:

Modify appModule
Instead of creating a new MainApplication instance, provide the application context that is passed to the Koin setup:

kotlin
Copy code
val appModule = module {
single { androidApplication() as MainApplication }
}